File: ts-sdk/src/sdk/models/operations/getweblinksid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetWebLinksIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=web_link_id" })
  webLinkId: string;
}


export class GetWebLinksIdHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=boxapi" })
  boxapi?: string;
}


export class GetWebLinksIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetWebLinksIdPathParams;

  @SpeakeasyMetadata()
  headers: GetWebLinksIdHeaders;
}


export class GetWebLinksIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  webLink?: shared.WebLink;
}

File: sdks/go/pkg/models/operations/getweblinksid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetWebLinksIDPathParams struct {
	WebLinkID string `pathParam:"style=simple,explode=false,name=web_link_id"`
}

type GetWebLinksIDHeaders struct {
	Boxapi *string `header:"style=simple,explode=false,name=boxapi"`
}

type GetWebLinksIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetWebLinksIDPathParams
	Headers    GetWebLinksIDHeaders
}

type GetWebLinksIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
	WebLink     *shared.WebLink
}
